tests/unittests/test_request_timeouts.py

```diff
@@ -0,0 +1,134 @@
+from tap_mailchimp.client import MailchimpClient
+import unittest
+from unittest.mock import patch
+import requests
+
+REQUEST_TIMEOUT_INT = 300
+REQUEST_TIMEOUT_STR = "300"
+REQUEST_TIMEOUT_FLOAT = 300.0
+
+# Mock response object
+def get_mock_http_response(*args, **kwargs):
+    contents = '{"access_token": "test", "expires_in":100, "accounts":[{"id": 12}]}'
+    response = requests.Response()
+    response.status_code = 200
+    response._content = contents.encode()
+    return response
+
+@patch("time.sleep")
+@patch("requests.Session.request", side_effect=requests.exceptions.Timeout)
+class TestRequestTimeoutsBackoff(unittest.TestCase):
+    
+    def test_no_request_timeout_in_config(self, mocked_request, mock_sleep):
+        """
+            Verify that if request_timeout is not provided in config then default value is used
+        """
+        # Initialize MailchimpClient object
+        client = MailchimpClient({'access_token': 'as'})
+        try:
+            client.request('GET', "http://test", "base_url")
+        except requests.exceptions.Timeout:
+            pass
+
+        # Verify that requests.Session.request is called 5 times
+        self.assertEqual(mocked_request.call_count, 5)
```

Comment:
@prijendev the test case name and the assertion that you are making is different. Can you please verify it?

Reply:
Yes, updated test case name.